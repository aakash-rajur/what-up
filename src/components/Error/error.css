.error-container {
    width: fit-content;
    max-width: 100%;
    display: inline-flex;
    flex-direction: column;
    align-items: center;
}

.error-container img {
    width: 52px;
    height: 52px;
}

.error-container span {
    text-align: center;
    font-size: 0.7em;
    font-weight: 100;
    letter-spacing: 0.2em;
}

.error-container pre {
    white-space: pre-wrap;
    text-align: center;
    font-weight: 100;
    line-height: 1.5em;
    font-size: 0.9em;
    padding: 0 1em;
}
