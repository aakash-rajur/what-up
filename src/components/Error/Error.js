import PropTypes from "prop-types";
import React from "react";
import ERROR_ICON from "../../assets/alert.svg";
import "./error.css";

function Error({text, className}) {
  return (
    <div className={`error-container ${className || ""}`}>
      <img src={ERROR_ICON} alt={text} />
      <pre>{text}</pre>
    </div>
  );
}

Error.propTypes = {
  text: PropTypes.string,
  className: PropTypes.string
};

Error.defaultProps = {
  text: "Something Went Wrong"
};

export default Error;
