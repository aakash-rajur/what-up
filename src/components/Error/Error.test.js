import React from 'react';
import ERROR_ICON from '../../assets/alert.svg';
import Error from './Error'

describe('Error component without error text', () => {
	let error = 'Something Went Wrong',
		component = shallow(<Error/>);
	it('should render', () => {
		expect(component.find('div.error-container').exists())
			.toBe(true);
	});
	
	it('dom tree integrity', checkDOM(component, [{
		selector: 'div.error-container',
		props: {
			className: 'error-container '
		},
		children: [{
			selector: 'img',
			length: 1
		}, {
			selector: 'pre',
			length: 1
		}]
	}, {
		selector: 'img',
		props: {
			src: ERROR_ICON,
			alt: error
		}
	}, {
		selector: 'pre',
		text: error
	}]));
	
	matchSnapshot(component)
});

describe('Error component with error text', () => {
	let error = 'hello world',
		component = shallow(<Error text={error}/>);
	
	component.setProps({text: error});
	
	it('dom tree integrity', checkDOM(component, [{
		selector: 'div.error-container',
		props: {
			className: 'error-container '
		},
		children: [{
			selector: 'img',
			length: 1
		}, {
			selector: 'pre',
			length: 1
		}]
	}, {
		selector: 'img',
		props: {
			src: ERROR_ICON,
			alt: error
		}
	}, {
		selector: 'pre',
		text: error
	}]));
	
	matchSnapshot(component);
});
