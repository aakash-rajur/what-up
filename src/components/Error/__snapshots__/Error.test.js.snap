// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component with error text match snapshot 1`] = `
<div
  className="error-container "
>
  <img
    alt="hello world"
    src="alert.svg"
  />
  <pre>
    hello world
  </pre>
</div>
`;

exports[`Error component without error text match snapshot 1`] = `
<div
  className="error-container "
>
  <img
    alt="Something Went Wrong"
    src="alert.svg"
  />
  <pre>
    Something Went Wrong
  </pre>
</div>
`;
